### ISSUE TITLE
Pi0 Fast version has large loss number during trainings

### ISSUE THREAD

**Issue by 1g0rrr** [Physical-Intelligence/openpi#418]
The fast version of Pi0 on our own robot starts training with a huge loss of 9.5.
After 12k steps, the loss is still 0.3. (I didn't try to run on the real robot.)
While the regular pi0 version has loss of 0.01 after the 12k steps on the same dataset. And works fine on real robot. 
<img width="1072" alt="Image" src="https://github.com/user-attachments/assets/8268bb1e-7d25-4bd8-83f0-574c667fe5f2" />

When I just run the default config pi0_fast_libero, it also starts with a huge loss number. So, the problem does not seem to be with our custom robot.
<img width="512" alt="Image" src="https://github.com/user-attachments/assets/1027aed5-8ec2-42cb-ad71-7b93a1195931" />

Are these supposed numbers?

**Comment by kpertsch** [comment]
Hi!

Please note that the loss values between pi0 and pi0_fast are not comparable at all. For pi0, the training objective is a MSE diffusion objective, while for pi0_fast it's a cross-entropy loss over discrete action tokens. So it is expected that the losses are very different, and I think your pi0_fast loss does not seem unusual.

If you want a more comparable metric, you can run the models over some training inputs and *decode* the continuous robot actions (ie for pi0_fast this means running the de-tokenizer, like you would do at inference time) and then compute the MSE between the decoded actions and the ground truth actions. This value would be comparable between both policies and could give you an indication whether the pi0_fast model learned to fit the data well.

It can also be helpful to just visualize some of the predictions in a line plot that shows the groudtruth action values for each dimension and the predicted action chunks so you can visually see whether they differ much.

Hope this helps to clarify things, will close the issue for now!
